assets/js/modules/analytics-4/components/widgets/MostEngagingPagesWidget.stories.js

```diff
@@ -0,0 +1,342 @@
+/**
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { MODULES_ANALYTICS_4 } from '../../datastore/constants';
+import {
+	provideKeyMetrics,
+	provideModuleRegistrations,
+	provideModules,
+} from '../../../../../../tests/js/utils';
+import { withWidgetComponentProps } from '../../../../googlesitekit/widgets/util';
+import {
+	STRATEGY_ZIP,
+	getAnalytics4MockResponse,
+	provideAnalytics4MockReport,
+} from '../../utils/data-mock';
+import { replaceValuesInAnalytics4ReportWithZeroData } from '../../../../../../.storybook/utils/zeroReports';
+import WithRegistrySetup from '../../../../../../tests/js/WithRegistrySetup';
+import { Provider as ViewContextProvider } from '../../../../components/Root/ViewContextContext';
+import { VIEW_CONTEXT_MAIN_DASHBOARD } from '../../../../googlesitekit/constants';
+import MostEngagingPagesWidget from './MostEngagingPagesWidget';
+import { ERROR_REASON_INSUFFICIENT_PERMISSIONS } from '../../../../util/errors';
+import { MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
+
+const pageViewsReportOptions = {
+	startDate: '2020-08-11',
+	endDate: '2020-09-07',
+	dimensions: [ 'pagePath' ],
+	metrics: [ { name: 'screenPageViews' } ],
+};
+
+const pageTitlesReportOptions = {
+	startDate: '2020-08-11',
+	endDate: '2020-09-07',
+	dimensionFilters: {
+		pagePath: new Array( 3 )
+			.fill( '' )
+			.map( ( _, i ) => `/test-post-${ i + 1 }/` )
+			.sort(),
+	},
+	dimensions: [ 'pagePath', 'pageTitle' ],
+	metrics: [ { name: 'screenPageViews' } ],
+	orderby: [ { metric: { metricName: 'screenPageViews' }, desc: true } ],
+	limit: 15,
+};
+
+const WidgetWithComponentProps = withWidgetComponentProps( 'test' )(
+	MostEngagingPagesWidget
+);
+
+const Template = ( { setupRegistry, viewContext, ...args } ) => (
+	<WithRegistrySetup func={ setupRegistry }>
+		<ViewContextProvider
+			value={ viewContext || VIEW_CONTEXT_MAIN_DASHBOARD }
+		>
+			<WidgetWithComponentProps { ...args } />
+		</ViewContextProvider>
+	</WithRegistrySetup>
+);
+
+export const Ready = Template.bind( {} );
```

Comment:
I think all the existing metric tiles with the tile table format have a "Ready View Only" story as well. Do you think it's worth adding that here as well?

Reply:
Good catch @nfmohit , I will add said story and request a re-review

Reply:
@nfmohit I have addressed the above, please see https://github.com/google/site-kit-wp/pull/7660#issuecomment-1750126935